api/v1alpha1/warehouse_types.go

```diff
@@ -104,7 +104,7 @@ type GitSubscription struct {
 	// URL is the repository's URL. This is a required field.
 	//
 	// +kubebuilder:validation:MinLength=1
-	// +kubebuilder:validation:Pattern=`^https?://(\w+([\.-]\w+)*@)?\w+([\.-]\w+)*(:[\d]+)?(/.*)?$`
+	// +kubebuilder:validation:Pattern=`^(https?://)?(\w+([\.-]\w+)*@)?\w+([\.-]\w+)*(:[\d\w-]+)?(/.*)?$`
```

Comment:
After tinkering with it a bit, I believe this will be a more complete regex:

```suggestion
	// +kubebuilder:validation:Pattern=`(?:^(https?)://(?:([\w-]+):(.+)@)?([\w-]+(?:\.[\w-]+)*)(?::(\d{1,5}))?(/.*)$)|(?:^([\w-]+)@([\w+]+(?:\.[\w-]+)*):(/?.*))`
```

https://regex101.com/r/DuAJs7/1

Reply:
Codegen would need to be re-run after this change.

Reply:
This regex also needs to be changed on the `RepoURL` field of the `GitRepoUpdate` type.

Reply:
Thanks, I completely missed that.